### DISCUSSION TITLE
Do I need to manually open files when cmd+clicking them?

### DISCUSSION THREAD

**Discussion by invm** [TypeFox/monaco-languageclient#701] [Q&A] [answered]
Hello amazing people.

I'm working from the python example and changed it to terraform field, I've managed to get lsp working and manually import tm grammar and language config. I have 2 files on the file system, peeking into the second file works as expected but when I cmd+click on it I see the code from the `useOpenEditorStub` function and nothing happens.

Do I need to open the file manually? If so how can it be achieved from that stub?

**Comment by kaisalmen** [comment, marked as answer]
Hi @invm `useOpenEditorStub` is really a stub. It does nothing.

You can do it like [here](https://github.com/CodinGame/monaco-vscode-api/blob/main/demo/src/setup.views.ts#L71). The function is defined [in this file](https://github.com/CodinGame/monaco-vscode-api/blob/main/demo/src/features/editor.ts).
